src/components/bg.vue

```diff
@@ -57,52 +89,57 @@ export default {
           }
         })
         .then(response => {
-          // console.log('[USP] unsplash response', response.body);
           console.log('[USP] got new image');
           const img = response.body;
 
-          localStorage.BG = img.urls.full + '&w=' + (window.innerWidth * window.devicePixelRatio); // Fix for width
-          this.BG = localStorage.BG;
+          localStorage.nextBG = img.urls.full + '&w=' + (window.innerWidth * window.devicePixelRatio); // Fix for width
+          this.nextBG = localStorage.nextBG;
 
-          this.imgColor = img.color;
-          localStorage.imgColor = img.color;
+          this.nextImgColor = img.color;
+          localStorage.nextImgColor = img.color;
 
           let author = {
             name: img.user.name,
             link: img.links.html
           };
-          this.author = author;
-          localStorage.author = JSON.stringify(author);
+          this.nextAuthor = author;
+          localStorage.nextAuthor = JSON.stringify(author);
+
+          localStorage.nextImage = img.id
+
+          this.cacheBG(this.nextBG);
+
+          // If the current bg still hasnt loaded by now, use this one (initial install)
+          // There may be a more elegant way to do this
+          if (!localStorage.BG) {
+              this.swapImages();
+          }
 
-          localStorage.image = img.id
-          localStorage.tsUpdatedImage = new Date().getTime();
         }, err => {
           // error callback
           console.error('[USP] Unsplash:', err.body);
-          delete localStorage.clear();
-
-          // fallback
-          this.BG = 'https://source.unsplash.com/random';
-          localStorage.BG = 'https://source.unsplash.com/random';
+          this.getCurrentImage(); // TODO: default to cached image if unsplash api quota is reached
         });
+
+
     },
     loadedImg() {
       console.timeEnd('BG');
       this.bgLoaded = true;
       console.log('[USP] 🏞 loaded from', this.BG.indexOf('data:') === 0 ? 'cache' : 'source');
       console.info('[USP] done. all set!');
       if (this.BG.indexOf('data:') !== 0) {
-        this.cacheIt();
+        this.cacheBG(this.BG);
       }
     },
-    cacheIt() {
-      console.info('[USP] Lets cacheIt...');
+    cacheBG(cacheBG) {
+      console.info('[USP] Lets cache the BG: ', cacheBG);
       var vm = this;
-      if (this.BG.indexOf('data') === -1) {
-        this.$http.get(this.BG).then(response => {
+      if (cacheBG.indexOf('data') === -1) {
+        this.$http.get(cacheBG).then(response => {
```

Comment:
Can you change this line to `this.$http.get(cacheBG, { responseType: 'blob' }).then(response => {`?
This will solve the error of caching the image.